### PR TITLE
Update core dependency and adopt to API changes

### PR DESCRIPTION
**Pull request by dmitryax**
Apply changes for https://github.com/open-telemetry/opentelemetry-collector/pull/6045

### CODE REVIEW

File: exporter/elasticsearchexporter/internal/objmodel/objmodel_test.go [hunk 1]
```diff
@@ -40,37 +40,45 @@ func TestObjectModel_CreateMap(t *testing.T) {
...
 			},
 			want: Document{[]field{{"i", IntValue(42)}, {"str", StringValue("test")}}},
 		},
 		"ignores nil values": {
 			build: func() Document {
-				return DocumentFromAttributes(pcommon.NewMapFromRaw(map[string]interface{}{
+				m := pcommon.NewMap()
+				m.FromRaw(map[string]interface{}{
 					"null": nil,
 					"str":  "test",
```
Comment (bogdandrutu) [inline exporter/elasticsearchexporter/internal/objmodel/objmodel_test.go:57]: Separate PR, maybe use `UpsertX` for the values?

File: internal/coreinternal/attraction/attraction_test.go [hunk 2]
```diff
@@ -40,10 +40,12 @@ type testCase struct {
...
-		attrMap := pcommon.NewMapFromRaw(tt.inputAttributes)
-		ap.Process(context.TODO(), nil, attrMap)
-		attrMap.Sort()
-		require.Equal(t, pcommon.NewMapFromRaw(tt.expectedAttributes).Sort(), attrMap)
+		inputMap := pcommon.NewMap()
+		inputMap.FromRaw(tt.inputAttributes)
+		ap.Process(context.TODO(), nil, inputMap)
+		expectedMap := pcommon.NewMap()
+		expectedMap.FromRaw(tt.expectedAttributes)
+		require.Equal(t, expectedMap.Sort(), inputMap.Sort())
```
Comment (bogdandrutu) [inline internal/coreinternal/attraction/attraction_test.go:48]:
```suggestion
		require.Equal(t, tt.expectedAttributes, inputMap.AsRaw())
```

Comment (dmitryax) [reply internal/coreinternal/attraction/attraction_test.go:48]: I wanted to do that but it requires all int values to be int64 :) I'll update it

File: exporter/elasticsearchexporter/internal/objmodel/objmodel_test.go (hunk 1 above)
Comment (dmitryax) [reply exporter/elasticsearchexporter/internal/objmodel/objmodel_test.go]: Makes sense, updated here since it's still related

**Review by bogdandrutu — APPROVED** [review]

**Pull request merged**